bundles/org.openhab.binding.nobohub/src/main/java/org/openhab/binding/nobohub/internal/connection/HubCommunicationThread.java

```diff
@@ -52,49 +87,78 @@ public void stopNow() {
     @Override
     public void run() {
         while (!stopped) {
-            try {
-                if (hubConnection.hasData()) {
-                    hubConnection.processReads(timeout);
-                }
-
-                if (Instant.now().isAfter(lastTimeFullScan.plus(NoboHubBindingConstants.TIME_BETWEEN_FULL_SCANS))) {
-                    hubConnection.refreshAll();
-                    lastTimeFullScan = Instant.now();
-                } else {
-                    hubConnection.handshake();
-                }
-
-                lastTimeReadStart = Instant.now();
-                hubConnection.processReads(timeout);
-            } catch (NoboCommunicationException nce) {
-                logger.error("Communication error with Hub", nce);
-                try {
-                    Duration readTime = Duration.between(Instant.now(), lastTimeReadStart);
-                    Thread.sleep(NoboHubBindingConstants.TIME_BETWEEN_RETRIES_ON_ERROR.minus(readTime).toMillis());
+            switch (currentState) {
+                case STARTING:
                     try {
-                        logger.debug("Trying to do a hard reconnect");
-                        hubConnection.hardReconnect();
-                    } catch (NoboCommunicationException nce2) {
-                        logger.debug("Failed to reconnect connection", nce2);
+                        hubConnection.refreshAll();
+                        lastTimeFullScan = Instant.now();
+                        setNextState(HubCommunicationThreadState.CONNECTED);
+                    } catch (NoboCommunicationException nce) {
+                        logger.error("Communication error with Hub", nce);
+                        setNextState(HubCommunicationThreadState.DISCONNECTED);
                     }
-                } catch (InterruptedException ie) {
-                    logger.debug("Interrupted from sleep after error");
-                    Thread.currentThread().interrupt();
-                }
+                    break;
+
+                case CONNECTED:
+                    try {
+                        if (hubConnection.hasData()) {
+                            hubConnection.processReads(timeout);
+                        }
+
+                        if (Instant.now()
+                                .isAfter(lastTimeFullScan.plus(NoboHubBindingConstants.TIME_BETWEEN_FULL_SCANS))) {
+                            hubConnection.refreshAll();
+                            lastTimeFullScan = Instant.now();
+                        } else {
+                            hubConnection.handshake();
+                        }
+
+                        hubConnection.processReads(timeout);
+                    } catch (NoboCommunicationException nce) {
+                        logger.error("Communication error with Hub", nce);
```

Comment:
Is this error unrecoverable and requires immediate action from the user? Otherwise log level should probably be reduced. See https://www.openhab.org/docs/developer/guidelines.html#f-logging.